mongo/integration/unified/matches.go

```diff
@@ -26,6 +26,16 @@ type keyPathCtxKey struct{}
 // if the value for this key is true.
 type extraKeysAllowedCtxKey struct{}
 
+// keypath is used to type paths within the structure of a document.  This type will help avoid using string-literals in
+// key-dependent logic, such as comparing strings to documents.
+type keypath string
```

Comment:
IMO, your approach here is unexpectedly abstract. Do you think there will be future cases for mixed type comparison in the unified test runner beyond the `comment` field?